src/lib/support/CommonPersistentData.h

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Contains a class handling creation of linked list of stored persistent data.
+ */
+
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/PersistentData.h>
+
+namespace chip {
+namespace CommonPersistentData {
+
+constexpr uint8_t kdefaultUndefinedEntry = 0;
+
+/// @brief Generic class to implement storage of a list persistently
+/// @tparam EntryType : Type of entry depends on the stored data
+/// @tparam kMaxSerializedSize : inherited from PersistentData class
+template <typename EntryType, size_t kMaxSerializedSize>
+struct StoredDataList : public PersistentData<kMaxSerializedSize>
+{
+    static constexpr TLV::Tag TagFirstEntry() { return TLV::ContextTag(1); }
```

Comment:
could this be `kTagFirstEntry = TLV::ContextTag(1)` i.e. a constant instead of method call?